controllers/gitrepository_controller.go

```diff
@@ -427,8 +427,13 @@ func (r *GitRepositoryReconciler) reconcileStorage(ctx context.Context,
 // change, it short-circuits the whole reconciliation with an early return.
 func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 	obj *sourcev1.GitRepository, commit *git.Commit, includes *artifactSet, dir string) (sreconcile.Result, error) {
+	gitImplementation := obj.Spec.GitImplementation
+	if enabled, _ := r.features[features.ForceGoGitImplementation]; enabled {
+		gitImplementation = sourcev1.GoGitImplementation
+	}
```

Comment:
As this is done multiple times, I am wondering if it might be better to factor this out into a helper function which takes the object and string map and returns the correct implementation?

Reply:
@darkowlzz gave a similar feedback on this and suggested the output of the feature gate to be passed as parameter to `gitCheckout`, which aligns with the current implementation for optimized clones. PTAL